File: Fundamentals_of_Software_Design/1_Basic_skills/Значение многочлена/main.py
def polynom(x, a, n):
    result = 0.0
    for i in range(n + 1):
        result += a[i] * (x ** i)  # Добавляем a[i] * x^i
    return result


# Пример использования
coefficients = [1, -3, 2]  # Многочлен: 2x^2 - 3x + 1
x_value = 3
degree = 2  # Степень многочлена
print(f"Значение многочлена: {polynom(x_value, coefficients, degree)}")  # 2*3^2 - 3*3 + 1 = 10

File: Fundamentals_of_Software_Design/1_Basic_skills/Значение многочлена/main.c
#include <stdio.h>

double polynom(double x, double a[], int n) {
    double result = 0.0;

    for (int i = n; i >= 0; i--) {
        result = result * x + a[i];
    }

    return result;
}

// Пример использования
int main() {
    double coefficients[] = {1, -2, 3, 4}; // Многочлен 4x^3 + 3x^2 - 2x + 1
    int degree = 3; // Степень многочлена
    double x = 2.0; // Точка, в которой вычисляется значение

    double value = polynom(x, coefficients, degree);
    printf("Значение многочлена в точке x = %.2f: %.2f\n", x, value);

    return 0;
}
